pkg/chains/storage/grafeas/grafeas.go

```diff
@@ -0,0 +1,378 @@
+/*
+Copyright 2020 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grafeas
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	attestationpb "github.com/grafeas/grafeas/proto/v1beta1/attestation_go_proto"
+	commonpb "github.com/grafeas/grafeas/proto/v1beta1/common_go_proto"
+	pb "github.com/grafeas/grafeas/proto/v1beta1/grafeas_go_proto"
+	"github.com/pkg/errors"
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/formats"
+	"github.com/tektoncd/chains/pkg/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	StorageBackendGrafeas = "grafeas"
+	projectNameFormat     = "projects/%s"
+	noteNameFormat        = "projects/%s/notes/%s"
+)
+
+// Backend is a storage backend that stores signed payloads in the storage that
+// is built on the top of grafeas i.e. container analysis.
+type Backend struct {
+	logger *zap.SugaredLogger
+	tr     *v1beta1.TaskRun
+	client pb.GrafeasV1Beta1Client
+	cfg    config.Config
+}
+
+// NewStorageBackend returns a new Grafeas StorageBackend that stores signatures in a Grafeas server
+func NewStorageBackend(ctx context.Context, logger *zap.SugaredLogger, tr *v1beta1.TaskRun, cfg config.Config) (*Backend, error) {
+	// build connection through grpc
+	// implicit uses Application Default Credentials to authenticate.
+	// Requires `gcloud auth application-default login` to work locally
+	creds, err := oauth.NewApplicationDefault(ctx, "https://www.googleapis.com/auth/cloud-platform")
+	if err != nil {
+		return nil, err
+	}
+
+	server := "dns:///containeranalysis.googleapis.com"
```

Comment:
We should document this behavior in the README.

Reply:
For sure, it seems like readme file doesn't have a dedicated section about storage backends or configuration. I am thinking about adding config docs in [this markdown file](https://github.com/tektoncd/chains/blob/main/docs/config.md#storage-configuration). Sounds good?